### DISCUSSION TITLE
How to define schema for Map<String, Any>?

### DISCUSSION THREAD

**Discussion by WasiqB** [json-schema-org/community#770] [Q&A]
I have a JSON file where there is one property called `capabilities`, which is a Map with String keys and it's value can be of any type, whether `String`, `Number` or `Boolean`.

Can anyone help me, how I can define it's schema correctly.

Example data:

```json
{
  "capabilities": {
    "appiumLogs": true,
    "appiumVersion": "2.0.0",
    "automationVersion": "latest",
    "buildName": "Test BrowserStack Build",
    "debug": true,
    "deviceLogs": true,
    "networkLogs": true,
    "projectName": "BrowserStack Android Project",
    "sessionName": "Test BrowserStack Session",
    "video": true
  }
}
```

I am trying to add my JSON files schema to the SchemaStore repository, but since I am not able to define correct schema for this property, I am blocked.

I tried the following schema for `capabilities`:

```json
{
  "type": "object",
  "additionalProperties": {
    "anyOf": [
      {
        "type": "string"
      },
      {
        "type": "number"
      },
      {
        "type": "boolean"
      }
    ]
  }
}
```

But this schema is failing the tests by giving the following message:

```shell
################ Error message
>> failed positive test | src/test/boyka-config/boyka-config.json
>> (Schema file) keywordLocation: #/properties/ui/properties/web/additionalProperties/properties/capabilities/additionalProperties/type
>> (Test file) instanceLocation:  /ui/web/test_lambda_test_chrome/capabilities/console
>> (Message)  must be string
>> Error in positive test.
##############################
```

Any help would be highly appreciated.

**Comment by notEthan** [comment]
To allow any type, you can simply not specify a type. So for an object whose properties may be any type, a schema of `{"type": "object"}` is sufficient.

**Comment by suprith-hub** [comment]
@WasiqB  You are looking for this maybe as per the example data:
```
{
  "properties":{
    "capabilities":{
      "additionalProperties": {"type": ["string", "boolean"] }
    }
  }
}
```
Having array of type will automatically mean any of those mentioned
